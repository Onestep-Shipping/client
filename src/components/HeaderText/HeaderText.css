.header-text-container {
  padding-bottom: 10px;
  cursor: pointer;
}

.header-text {
  font-size: 20px;
  color: white;
  cursor: pointer;
  user-select: none;
  margin: 0px 10px;
}

.header-text-container:hover {
  border-bottom: solid white 4px;
}