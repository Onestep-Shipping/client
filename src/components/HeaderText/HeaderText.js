import './HeaderText.css';

import PropTypes from 'prop-types';
import React from 'react';

const HeaderText = (props) => {
  const {value, action} = props;

  return (
    <div className="header-text-container" onClick={action}>
      <span className="header-text">{value}</span>
    </div>
  );
};

export default HeaderText;

HeaderText.propTypes = {
  value: PropTypes.string,
  action: PropTypes.func
};