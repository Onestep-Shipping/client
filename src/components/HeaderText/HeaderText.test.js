import HeaderText from './HeaderText';
import React from 'react';
import { shallow } from 'enzyme';

const value = "Contact";
const action = jest.fn();

let wrapper = shallow(
  <HeaderText value={value} action={action} />
);

describe('HeaderText', () => {
  it('should render the HeaderText Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
  it('should renders the props value', () => { 
    expect(wrapper.find('span').text()).toEqual(value);
  });
  it('should trigger the action when clicked', () => { 
    wrapper.simulate('click');
    expect(action).toHaveBeenCalled();
  });
});