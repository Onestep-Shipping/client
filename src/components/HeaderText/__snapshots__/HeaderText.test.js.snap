// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderText should render the HeaderText Component correctly 1`] = `
<div
  className="header-text-container"
  onClick={[MockFunction]}
>
  <span
    className="header-text"
  >
    Contact
  </span>
</div>
`;
